yapapi/executor/ctx.py

```diff
@@ -378,15 +408,18 @@ def download_json(
             _ReceiveJson(self._storage, src_path, on_download, limit, self._emitter)
         )
 
-    def commit(self, timeout: Optional[timedelta] = None) -> Work:
+    def commit(self, timeout: Optional[timedelta] = None, wait_for_results: bool = True) -> Work:
```

Comment:
I'd make this an internal interface and expose only the `commit_async` to users

```suggestion
    def commit(self, timeout: Optional[timedelta] = None, __wait_for_results: bool = True) -> Work:
```